scripts/update-index-alias.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+"""
+A script to update the search index an alias is assoiciated with.
+
+To delete old indexes (those with no alias), set --delete-old-indexes=yes
+
+Usage:
+scripts/update-index-alias.py <alias> <target> <stage> <search-api-endpoint> [options]
+
+Options:
+    --delete-old-indexes=<delete-old-indexes>  [default: ]
+"""
+import os
+import sys
+import subprocess
+import yaml
+import json
+import requests
+from requests.exceptions import HTTPError
+from docopt import docopt
+
+
+def update_index_alias(alias, target, stage, endpoint, delete_old_indexes):
+    url = "https://{}/{}".format(endpoint, alias)
+    auth_token = _get_auth_token(stage)
+    headers = {
+        'Authorization': "Bearer {}".format(auth_token),
+        'Content-type': 'application/json'
+    }
+    data = {'type': 'alias', 'target': "{}".format(target)}
+
+    response = requests.put(url, headers=headers, json=data)
+
+    try:
+        response.raise_for_status()
+    except HTTPError as e:
+        print("HTTPError updating alias: {}".format(e.args[0]))
+        sys.exit(1)
+    except Exception as e:
+        print("Error updating alias: {}".format(e))
+        sys.exit(2)
+
+    print('Successfully updated alias')
+
+    if delete_old_indexes == 'yes':
+        _delete_old_indexes(endpoint, auth_token)
+
+
+def _get_auth_token(stage):
+    DM_CREDENTIALS_REPO = os.environ.get('DM_CREDENTIALS_REPO')
+    creds = subprocess.check_output([
+        "{}/sops-wrapper".format(DM_CREDENTIALS_REPO),
+        "-d",
+        "{}/vars/{}.yaml".format(DM_CREDENTIALS_REPO, stage)
+    ])
+    return yaml.load(creds)['search_api']['auth_tokens'][2]
+
+
+def _delete_old_indexes(endpoint, auth_token):
```

Comment:
I still feel like deleting all old/un-aliased indexes probably falls outside the remit of this script, which at its core is only about switching over index aliases. If/When we start having more things indexed than just the G-Cloud catalogue, this could be a pretty destructive action with unintended/unanticipated consequences.

I'd rather have to manually delete errant indexes one-by-one (which are highly unlikely to occur in production, anyway) than have a minesweeper go over them all regularly and delete indiscriminately. I'm OK with this script deleting the one oldest index relevant to the current target index (in this instance G-Cloud), but it should probably leave everything else.